kolibri/logger/models.py

```diff
@@ -98,3 +100,56 @@ class UserSessionLog(BaseLogModel):
     start_timestamp = models.DateTimeField(auto_now_add=True)
     completion_timestamp = models.DateTimeField(blank=True, null=True)
     pages = models.TextField(blank=True)
+
+class MasteryLog(BaseLogModel):
+    """
+    This model provides a summary of a user's engagement with an assessment within a mastery level
+    """
+    # Every MasteryLog is related to the single summary log for the user/content pair
+    summarylog = models.ForeignKey(ContentSummaryLog, related_name="masterylogs")
+    # The MasteryLog records the mastery criterion that has been specified for the user.
+    # It is recorded here to prevent this changing in the middle of a user's engagement
+    # with an assessment.
+    mastery_criterion = models.CharField(max_length=200)
+    start_timestamp = models.DateTimeField()
+    end_timestamp = models.DateTimeField(blank=True, null=True)
```

Comment:
Since this is separate from `completion_timestamp`, and since it's the default sort-order field, we should have it be required. Every time we make changes and save, it would be set to the current time -- so it's really more of a "last interaction" timestamp, I guess?